### PR TITLE
Functional tests - Add new test 'FO > Hummingbird > CRUD address'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'FO > Hummingbird > CRUD address'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI and nightly are :green_apple: 
| UI Tests          | https://github.com/nesrineabdmouleh/testing_pr/actions/runs/7834384775
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/hummingbird/03_userAccount/02_CRUDAddress.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  let firstAddressPosition: number = 0;
+  let secondAddressPosition: number = 0;
+
+  const newCustomerData: CustomerData = new CustomerData();
+  const createAddressData: AddressData = new AddressData({country: 'France'});
+  const editAddressData: AddressData = new AddressData({country: 'France'});
+  const secondAddressData: AddressData = new AddressData({country: 'France'});
+
+  // Pre-condition
+  createAccountTest(newCustomerData, `${baseContext}_preTest_2`);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/hummingbird/03_userAccount/02_CRUDAddress.ts]:
```suggestion
  createAccountTest(newCustomerData, `${baseContext}_preTest_1`);
```

File: tests/UI/campaigns/functional/FO/hummingbird/03_userAccount/02_CRUDAddress.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+  // Pre-condition : Install Hummingbird
+  installHummingbird(`${baseContext}_preTest_3`);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/hummingbird/03_userAccount/02_CRUDAddress.ts]:
```suggestion
  installHummingbird(`${baseContext}_preTest_2`);
```

File: tests/UI/pages/FO/classic/cart/index.ts [hunk 3]
```diff
@@ -100,7 +100,7 @@ class CartPage extends FOBasePage {
 
   private readonly alertWarning: string;
 
-  private readonly proceedToCheckoutButton: string;
+  public proceedToCheckoutButton: string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/cart/index.ts]:
```suggestion
  protected proceedToCheckoutButton: string;
```

File: tests/UI/pages/FO/classic/checkout/index.ts [hunk 4]
```diff
@@ -211,7 +211,7 @@ class Checkout extends FOBasePage {
 
   private readonly deliveryAddressSection: string;
 
-  private readonly deliveryAddressPosition: (position: number) => string;
+  public deliveryAddressPosition: (position: number) => string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/checkout/index.ts]:
```suggestion
  protected deliveryAddressPosition: (position: number) => string;
```

File: tests/UI/pages/FO/classic/myAccount/addAddress.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ class AddAddressPage extends FOBasePage {
 
   public readonly updateFormTitle: string;
 
-  private readonly pageHeaderTitle: string;
+  public pageHeaderTitle: string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/myAccount/addAddress.ts]:
```suggestion
  protected pageHeaderTitle: string;
```

File: tests/UI/pages/FO/classic/myAccount/addresses.ts [hunk 6]
```diff
@@ -23,9 +23,9 @@ class AddressesPage extends FOBasePage {
 
   private readonly addressBlock: string;
 
-  private readonly addressBodyTitle: string;
+  public addressBodyTitle: string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/myAccount/addresses.ts]:
```suggestion
  protected addressBodyTitle: string;
```

File: tests/UI/pages/FO/classic/myAccount/addresses.ts [hunk 7]
```diff
@@ -23,9 +23,9 @@ class AddressesPage extends FOBasePage {
...
 
-  private readonly createNewAddressLink: string;
+  public createNewAddressLink: string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/myAccount/addresses.ts]:
```suggestion
  protected createNewAddressLink: string;
```

File: tests/UI/pages/FO/classic/product/index.ts [hunk 8]
```diff
@@ -49,7 +49,7 @@ class Product extends FOBasePage {
 
   private readonly blockCartModal: string;
 
-  private readonly proceedToCheckoutButton: string;
+  public proceedToCheckoutButton: string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/classic/product/index.ts]:
```suggestion
  protected proceedToCheckoutButton: string;
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks 
- No need to put properties in `public`, `protected` is OK

File: tests/UI/pages/FO/classic/product/index.ts (hunk 8 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/classic/product/index.ts]: Done thx :)

**Comment by nesrineabdmouleh** [comment]
New run => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/7817951169

**Review by Progi1984 — DISMISSED** [review]

**Review by SD1982 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Pull request merged**